File: scripts/debug_app.js
import os

app_js = r'C:\Users\Administrator\Desktop\iEXA-WIN\src\renderer\app.js'
with open(app_js, 'r', encoding='utf-8') as f:
    content = f.read()

print(f'File size: {len(content)}')

# Search for the exact string
target = 'data.models.length'
idx = content.find(target)
if idx >= 0:
    print(f'Found at char {idx}')
    print(repr(content[idx-50:idx+200]))
else:
    print('Not found')
    # Try finding just 'models.length'
    idx2 = content.find('models.length')
    if idx2 >= 0:
        print(f'models.length found at {idx2}')
        print(repr(content[idx2-100:idx2+200]))

File: scripts/debug_app.py
import os

app_js = r'C:\Users\Administrator\Desktop\iEXA-WIN\src\renderer\app.js'

with open(app_js, 'r', encoding='utf-8') as f:
    content = f.read()

print(f'File size: {len(content)}')

# Debug: find the data.models.map pattern
idx = content.find('data.models')
if idx >= 0:
    chunk = content[idx:idx+300]
    # Write to file instead of printing
    with open(r'C:\Users\Administrator\Desktop\iEXA-WIN\scripts\debug_app.txt', 'w', encoding='utf-8') as f:
        f.write(repr(chunk))
    print('Written to debug_app.txt')
else:
    print('data.models not found')

# Also check if the file was already patched
idx2 = content.find('modelList')
if idx2 >= 0:
    print(f'modelList found at {idx2}')